src/modules/expression-evaluator/src/evaluateExpression.test.ts

```diff
@@ -585,7 +558,29 @@ test('Test GraphQL -- count Sections on current Application', () => {
       endpoint: graphQLendpoint,
     },
   }).then((result: any) => {
-    expect(result).toEqual(3)
+    expect(result).toEqual(2)
+  })
+})
+
+test('Test GraphQL -- get continents list from External Graphql API ', () => {
+  return evaluateExpression(testData.GraphQL_GetContinentsList_ExternalAPI, {
+    graphQLConnection: {
+      fetch: fetch,
+      endpoint: graphQLendpoint,
+    },
+  }).then((result: any) => {
+    expect(result).toEqual(testData.continentsResult)
+  })
+})
+
+test('Test GraphQL -- get country details by code from External Graphql API ', () => {
+  return evaluateExpression(testData.GraphQL_GetCountryByCode_ExternalAPI, {
+    graphQLConnection: {
+      fetch: fetch,
+      endpoint: graphQLendpoint,
+    },
```

Comment:
The `graphQLConnection` object shouldn't be needed here, so it might be a good idea to remove it from _one_ of these tests to test for that case. (Like, if the GraphQL evaluator looks for a GraphQLConnection object and doesn't see one, it should still work, so we need to test for that)

Reply:
Actually, scratch that -- it's still needed for the `fetch` object. So maybe instead omit the `endpoint` property from the object on one of these and make sure it still works. (You'll probably need to make "endpoint" an optional property in the types).

Reply:
No, that's where it gets the `default` one from. As discussed in person, we could have changed that so front-end doesn't specify the default endpoint, but for now that's what is used for it. We can change that later, I don't think it's a priority for now. Maybe it can be a note to take as part of the Build to production Epic.

Reply:
Yeah, but if you're passing an external URL in the query expression, you should be able to omit the `endpoint` field in the `graphQLConnection` object that you pass in, and the evaluator should still work, hence we should make sure there is a test that handles that case.

(Use case: if using the evaluator for something else (i.e. not IRIMS), you might not even have a local GraphQL connection, so why require an endpoint field for external queries?)

Reply:
I'll set this as a separated issue, I feel this is extra and really would like to continue with the progress of getting the changes ready to interact with UC codes server. And this will be an improvement to be done - when needed.